snobal_1d/station_pipeline.py

```diff
@@ -0,0 +1,257 @@
+import MesoPy
+import json
+import pandas as pd
+import numpy as np
+import math
+import plotly.express as px
+from plotly.subplots import make_subplots
+import plotly.graph_objects as go
+import chart_studio.plotly as py
+
+# only needed if intending to pull station data again. 
+from MesoPy import Meso
+m = Meso(token='YOUR TOKEN HERE')
+
+# constants
+SB_CONST = 5.67e-8
+
+
+def html_chart(df, height=1200):
+    """
+    make interactive chart. 
+    
+    param df: inpute dataframe
+    param height: optional plot height
+    returns: plotly chart
+    """
+    fig = make_subplots(rows=(len(df.columns)), 
+                        cols=1, 
+                        subplot_titles=df.columns,
+                        shared_xaxes=True,
+                        vertical_spacing=0.007
+                        )
+    j = 1
+    for i in df.columns:
+        fig.add_trace(
+            go.Scatter(
+                {'x': df.index, 
+                 'y': df[i]}), 
+                 row=j, col=1)
+        j += 1
+    
+    fig.update_layout(height=height, font_size=9)
+    #py.plot(fig, filename = 'Atwater', auto_open=True)
+    return fig
+
+def vapor_pressure(df, dt):
+    """
+    Vapor pressure dervied from dewpoint temp. Formula from NWS: 
+    https://www.weather.gov/media/epz/wxcalc/vaporPressure.pdf
+    
+    :param df: dataframe with dewpoint temp
+    :param dt: column with dewpoint temp (in Deg C). 
+    :return: vapor pressure col in dataframe (output is in Pa).
+    """
+    df['vp'] = 6.11 * (10 ** ((7.5 * df[dt]) / (237.3 + df[dt])))
+    # hPa to Pa
+    df['vp'] = df['vp'] * 100
+    return df
+
+def longwave_est_2(RH_col, sw_net, air_t_col_K):
+    """
+    Updated longwave estimation from Immerzeel et al. 2019. Currently implemented
+    with list comprehension is Irwin_pipeline.ipynb, because I had issues getting 
+    multiple df args to work w/ a lambda function.
+    
+    :param RH_col: col with RH
+    :param sw_net: col with net shortwave, in W/M^2
+    :param air_t_col_K: col with 
+    :return: list populated with estimated longwave.
+    """
+    SB_CONST = 5.67e-8
```

Comment:
Very nice! Using all caps for constants is a good practice

Reply:
Looks like this one is duplicate though. Sure you need to declare this again?

Reply:
Not a duplicate; there are two different parameterizations. I will make it more clear in the docstring.

Reply:
Help me out:
Line 15: `SB_CONST = 5.67e-8`
and you are re-defining it here. Where is the difference?

Reply:
Ah, the constant. Fixed!